### PR TITLE
doc: add document for dm-migrate

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
ref: https://internal.pingcap.net/jira/browse/TOOL-511

### What is changed and how it works?
rewrite the docs about how to migrate dm-worker.

### Check List 

Tests 

- No code

### CODE REVIEW

**Comment by IANTHEREAL** [comment]
@amyangfei @csuzhangxc PTAL

File: docs/dm/zh_CN/README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ DM (Data Migration) 是一体化数据同步任务管理平台，支持全量备
         - [重启 DM 组件注意事项](./maintenance/caution-for-restart-dm.md)
         - [使用 dmctl 执行主-从切换](./maintenance/master-slave-switch.md)
         - [DM 监控与告警](./maintenance/metrics-alert.md)
+        - [迁移dm-worker](./maintenance/dm-migrate.md)
```
Comment (amyangfei) [inline docs/dm/zh_CN/README.md]:
```suggestion
        - [迁移 DM-worker](./maintenance/dm-migrate.md)
```

File: docs/dm/zh_CN/maintenance/dm-migrate.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+迁移dm-worker
```
Comment (amyangfei) [inline docs/dm/zh_CN/maintenance/dm-migrate.md]:
```suggestion
迁移 DM-worker
```

Comment (amyangfei) [reply docs/dm/zh_CN/maintenance/dm-migrate.md]: need to change all `dm-worker` to `DM-worker`, and add a space between Chinese character and `DM-worker`

File: docs/dm/zh_CN/maintenance/dm-migrate.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+当dm-worker不可用时，需要将在不可用的dm-worker上运行的任务迁移到另一台dm-worker上
+
+### 迁移流程
+1. 部署新的dm-worker
+2. 确定不可用的dm-worker同步点
+2. 对新部署的dm-worker使用`migrate-relay`命令，接上一个dm-worker的同步点
+
+### 如何确定不可用的dm-worker同步点
+通常会保存在下游数据库里，task配置里的`meta-schema`字段指名了保存在下游数据库的位置。
```
Comment (amyangfei) [inline docs/dm/zh_CN/maintenance/dm-migrate.md:12]: 可以具体举一个例子，譬如任务名是 `test`
详细说明从下游元数据库哪一张表 `test_syncer_checkpoint`，选取哪一行数据库的 `binlog_name` 和 `binlog_pos`
例如：`select binlog_name, binlog_pos from `dm_meta`.`test_syncer_checkpoint` where id = ? and is_global = 1`

**Comment by imtbkcat** [comment]
PTAL @amyangfei

File: docs/dm/zh_CN/README.md [hunk 4]
```diff
@@ -44,6 +44,7 @@ DM (Data Migration) 是一体化数据同步任务管理平台，支持全量备
         - [重启 DM 组件注意事项](./maintenance/caution-for-restart-dm.md)
         - [使用 dmctl 执行主-从切换](./maintenance/master-slave-switch.md)
         - [DM 监控与告警](./maintenance/metrics-alert.md)
+        - [迁移DM-worker](./maintenance/dm-migrate.md)
```
Comment (amyangfei) [inline docs/dm/zh_CN/README.md]:
```suggestion
        - [迁移 DM-worker](./maintenance/dm-migrate.md)
```

Comment (imtbkcat) [reply docs/dm/zh_CN/README.md]: @amyangfei addressed.

File: docs/dm/zh_CN/maintenance/dm-migrate.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+1. 部署新的 DM-worker
+2. 确定不可用的 DM-worker 同步点
+2. 对新部署的 DM-worker 使用 `migrate-relay` 命令，接上一个 DM-worker 的同步点
+
+### 如何确定不可用的 DM-worker 同步点
+通常会保存在下游数据库里，task配置里的`meta-schema`字段指名了保存在下游数据库的位置。
+
+这里假设`meta_schema`的值是`dm_meta`， 任务名是`test`就到下游数据库中找到`dm_meta`这个库中，
+找到`test_syncer_checkpoint`这个表, 用SQL查询出所有的同步点
+`select binlog_name, binlog_pos from dm_meta.test_syncer_checkpoint where id = 要迁移的 DM-worker 的id and is_global = 1;`
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/maintenance/dm-migrate.md]: `要迁移的 DM-worker 的id` or `要迁移的 DM-worker 对应的上游 MySQL 的 source-id`?

File: docs/dm/zh_CN/maintenance/dm-migrate.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+选取最近确认的同步点， 如果 DM-worker 上运行着多个任务, 需要找到所有任务的同步点，取其中时间最早的同步点。
+
+### 同步点包括哪些信息
+需要 binlog-name 和 binlog-pos
+参考[relay-log](../features/relay-log.md)
+
+### 命令参数解释
+`migrate-relay <worker> <binlogName> <binlogPos>`
+- `worker`: 新部署的 DM-worker 的全局ID
+- `binlogName`: 对应同步点的`binlog-name`
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/maintenance/dm-migrate.md]: is it needed to convert this `binlog-name`?
`mysql-bin|000002.000003` to `mysql-bin.000003`?

Comment (imtbkcat) [reply docs/dm/zh_CN/maintenance/dm-migrate.md]: @csuzhangxc addressed

File: docs/dm/zh_CN/maintenance/dm-migrate.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+1. 部署新的 DM-worker
+2. 确定不可用的 DM-worker 同步点
+2. 对新部署的 DM-worker 使用 `migrate-relay` 命令，接上一个 DM-worker 的同步点
+
+### 如何确定不可用的 DM-worker 同步点
+通常会保存在下游数据库里，task配置里的`meta-schema`字段指名了保存在下游数据库的位置。
+
+这里假设`meta_schema`的值是`dm_meta`， 任务名是`test`就到下游数据库中找到`dm_meta`这个库中，
+找到`test_syncer_checkpoint`这个表, 用SQL查询出所有的同步点
+`select binlog_name, binlog_pos from dm_meta.test_syncer_checkpoint where id = 要迁移的 DM-worker 对应的上游 MySQL 的 source-id and is_global = 1;`
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/maintenance/dm-migrate.md]: do you want to show this `SELECT` statement in a separate line? maybe you need to add spaces at the tail of the line above or use ` ``` `to start a code block?

in addition, use some marker to wrap `要迁移的 DM-worker 对应的上游 MySQL 的 source-id` like `{要迁移的 DM-worker 对应的上游 MySQL 的 source-id}`?

File: docs/dm/zh_CN/maintenance/dm-migrate.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+就要将其转换为`mysql-bin.000003`
+
+### 同步点包括哪些信息
+需要 binlog-name 和 binlog-pos
+参考[relay-log](../features/relay-log.md)
+
+### 命令参数解释
+`migrate-relay <worker> <binlogName> <binlogPos>`
+- `worker`: 新部署的 DM-worker 的全局ID
+- `binlogName`: 对应同步点的`binlog-name`
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/maintenance/dm-migrate.md]: s/`binlog-name`/`binlog_name`. unify names?
ditto for `binlog-pos`

**Review by csuzhangxc — DISMISSED** [review]

File: docs/dm/zh_CN/maintenance/dm-migrate.md (hunk 8 above)
Comment (imtbkcat) [reply docs/dm/zh_CN/maintenance/dm-migrate.md]: ok

**Comment by csuzhangxc** [comment]
LGTM

File: docs/dm/zh_CN/maintenance/dm-migrate.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+```
+select binlog_name, binlog_pos from dm_meta.test_syncer_checkpoint where id = { 要迁移的 DM-worker 对应的上游 MySQL 的 source-id } and is_global = 1;
+```
+选取最近确认的同步点， 如果 DM-worker 上运行着多个任务, 需要找到所有任务的同步点，取其中时间最早的同步点。
+
+需要将得到的`binlog_name`做一步转换。例如：得到的`binlog_name`是`mysql-bin|000002.000003`，
+就要将其转换为`mysql-bin.000003`
+
+### 同步点包括哪些信息
+需要 binlog-name 和 binlog-pos
```
Comment (amyangfei) [inline docs/dm/zh_CN/maintenance/dm-migrate.md]:
```suggestion
需要 binlog_name 和 binlog_pos
```

**Review by amyangfei — COMMENTED** [review]
Rest LGTM

File: docs/dm/zh_CN/maintenance/dm-migrate.md (hunk 9 above)
Comment (imtbkcat) [reply docs/dm/zh_CN/maintenance/dm-migrate.md]: @amyangfei addressed

**Review by amyangfei — COMMENTED** [review]
LGTM

**Comment by imtbkcat** [comment]
@amyangfei can you merge this pr, i dont have write access to this repository....

**Pull request merged**